src/frontend/src/components/SettingSidebarItem.js

```diff
@@ -0,0 +1,30 @@
+import React, {useEffect, useState} from "react";
+import {Link} from "react-router-dom";
+
+export const SidebarItem = ({to,icon,title})=>{
+
+    const [width,setWidth] = useState(window.innerWidth)
+
+    useEffect(()=>{
+        const handleWidth = ()=>{
+            setWidth(window.innerWidth)
+        }
+        window.addEventListener("resize",handleWidth)
+        return ()=>{
+            window.removeEventListener("resize",handleWidth)
+        }
+    })
```

Comment:
```suggestion
    const {width} = useWindowSize()
```
You can simply access the width using this hook now